Source/Core/Core/IOS/ES/ES.cpp

```diff
@@ -220,8 +222,8 @@ IPCReply ESDevice::GetTitleDirectory(const IOCtlVRequest& request)
   const u64 title_id = memory.Read_U64(request.in_vectors[0].address);
 
   char* path = reinterpret_cast<char*>(memory.GetPointer(request.io_vectors[0].address));
-  sprintf(path, "/title/%08x/%08x/data", static_cast<u32>(title_id >> 32),
-          static_cast<u32>(title_id));
+  fmt::format_to(path, "/title/{:08x}/{:08x}/data", static_cast<u32>(title_id >> 32),
+                 static_cast<u32>(title_id));
```

Comment:
GetPointer is fundamentally unsafe, as it makes no guarantees about how many bytes you're allowed to write. It would be better if this code formatted into a local buffer and then used CopyToEmu, like NetIPTopDevice::HandleInetNToPRequest does.

Reply:
Done. Let me know if I happened to handle it incorrectly.